src/rabbit_stomp_util.erl

```diff
@@ -260,6 +260,26 @@ msg_header_name("1.2") -> ?HEADER_ACK;
 msg_header_name("1.1") -> ?HEADER_MESSAGE_ID;
 msg_header_name("1.0") -> ?HEADER_MESSAGE_ID.
 
+build_arguments(Headers) ->
+    Arguments = lists:foldl(fun({K, V}, Acc) ->
+                                case lists:member(K, ?HEADER_ARGUMENTS) of
+                                  true  -> [build_argument(K, V) | Acc];
+                                  false -> Acc
+                                end
+                            end,
+                            [],
+                            Headers),
+    {arguments, Arguments}.
+
+%% build the actual value thru pattern matching
+build_argument(?HEADER_X_DEAD_LETTER_EXCHANGE, Val) -> {list_to_binary(?HEADER_X_DEAD_LETTER_EXCHANGE), longstr, list_to_binary(string:strip(Val))};
+build_argument(?HEADER_X_DEAD_LETTER_ROUTING_KEY, Val) -> {list_to_binary(?HEADER_X_DEAD_LETTER_ROUTING_KEY), longstr, list_to_binary(string:strip(Val))};
+build_argument(?HEADER_X_EXPIRES, Val) -> {list_to_binary(?HEADER_X_EXPIRES), long, list_to_integer(string:strip(Val))};
```

Comment:
Expiration is a string in the protocol, [the docs](http://www.rabbitmq.com/ttl.html) mention this.